File: daily/python/Nov-17-minimize-maximum-pair-sum-in-array.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        l = nums.size()
        res = 0
        for i in range(l//2):
            res = max(res, nums[i] + nums[l-i-1])
        return res

File: daily/cpp/Nov-17-minimize-maximum-pair-sum-in-array.cpp
#include <vector>

using namespace std;
class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int l = nums.size();
        int res = 0;
        for (int i = 0; i < l / 2; i++) {
            res = max(res, nums[i] + nums[l - i - 1]);
        }

        return res;
    }
    // ez w/ the sort, but with time complexity of O(nlogn)
}